libse/Forms/SplitLongLinesHelper.cs

```diff
@@ -43,105 +43,113 @@ public static bool QualifiesForSplit(string text, int singleLineMaxCharacters, i
         public static Subtitle SplitLongLinesInSubtitle(Subtitle subtitle, int totalLineMaxCharacters, int singleLineMaxCharacters)
         {
             var splittedSubtitle = new Subtitle(subtitle);
-            splittedSubtitle.Paragraphs.Clear();
             string language = LanguageAutoDetect.AutoDetectGoogleLanguage(subtitle);
-            for (int i = 0; i < subtitle.Paragraphs.Count; i++)
+
+            // calculate gaps
+            var halfMinGaps = Configuration.Settings.General.MinimumMillisecondsBetweenLines / 2.0;
+            var halfMinGapsMood = halfMinGaps + Configuration.Settings.General.MinimumMillisecondsBetweenLines % 2;
+
+            const int FirstLine = 0;
+            const int SecondLine = 1;
+
+            for (int i = splittedSubtitle.Paragraphs.Count - 1; i >= 0; i--)
             {
-                bool added = false;
-                var p = subtitle.GetParagraphOrDefault(i);
-                if (p?.Text != null)
+                var oldParagraph = splittedSubtitle.Paragraphs[i];
+
+                // don't split into two paragraph if it can be balanced
+                var text = Utilities.AutoBreakLine(oldParagraph.Text, language);
+                if (!QualifiesForSplit(text, singleLineMaxCharacters, totalLineMaxCharacters))
                 {
-                    if (QualifiesForSplit(p.Text, singleLineMaxCharacters, totalLineMaxCharacters))
-                    {
-                        var text = Utilities.AutoBreakLine(p.Text, language);
-                        if (!QualifiesForSplit(text, singleLineMaxCharacters, totalLineMaxCharacters))
-                        {
-                            var newParagraph = new Paragraph(p) { Text = text };
-                            splittedSubtitle.Paragraphs.Add(newParagraph);
-                            added = true;
-                        }
-                        else
-                        {
-                            if (text.Contains(Environment.NewLine))
-                            {
-                                var arr = text.SplitToLines();
-                                if (arr.Count == 2)
-                                {
-                                    var minMsBtwnLnBy2 = Configuration.Settings.General.MinimumMillisecondsBetweenLines / 2;
-                                    int spacing1 = minMsBtwnLnBy2;
-                                    int spacing2 = minMsBtwnLnBy2;
-                                    if (Configuration.Settings.General.MinimumMillisecondsBetweenLines % 2 == 1)
-                                    {
-                                        spacing2++;
-                                    }
-
-                                    double duration = p.Duration.TotalMilliseconds / 2.0;
```

Comment:
I think this is really wrong, we can't assume both line with have same about of characters... this problem is handle in this PR.